src/ast/passes/semantic_analyser.cpp

```diff
@@ -216,6 +216,303 @@ class SemanticAnalyser : public Visitor<SemanticAnalyser> {
 
 } // namespace
 
+struct assign_checks {
+  bool want_map = false;
+  bool want_var = false;
+  bool want_map_key = false;
+};
+
+// clang-format off
+static const std::map<std::string, assign_checks> CALL_ASSIGN_SPEC = {
+  { "avg", { .want_map = true } },
+  { "cat", {} },
+  { "clear", {} },
+  { "count", { .want_map = true } },
+  { "debugf", {} },
+  { "delete", {} },
+  { "exit", {} },
+  { "hist", { .want_map = true } },
+  { "join", {} },
+  { "lhist", { .want_map = true } },
+  { "max", { .want_map = true } },
+  { "min", { .want_map = true } },
+  { "override", {} },
+  { "print", {} },
+  { "printf", {} },
+  { "signal", {} },
+  { "skboutput", {.want_var = true} },
+  { "stats", { .want_map = true } },
+  { "sum", { .want_map = true } },
+  { "system", {} },
+  { "time", {} },
+  { "zero", {} },
+};
+
+struct arg_type_spec {
+  Type type = Type::integer;
+  bool literal = false;
+  // This means that this check is only valid if there are
+  // this exact number of args in the call
+  size_t nargs = 0;
+  // This indicates that this is just a placeholder
+  // as we use the index in the vector of arg_type_spec
+  // as the number argument to check
+  bool no_check = false;
+};
+
+struct arg_spec {
+  size_t min;
+  size_t max;
+  // NOLINTBEGIN(readability-redundant-member-init)
+  std::vector<arg_type_spec> types = {};
+  // NOLINTEND(readability-redundant-member-init)
```

Comment:
Is the initializer that important that using it is worth the `NOLINTBEGIN`/`NOLINTEND` lines?

Reply:
:facepalm: